js/upload/src/utils.ts

```diff
@@ -65,69 +65,99 @@ function is_url(str: string): boolean {
 
 export function get_fetchable_url_or_file(
 	path: string | null,
-	root: string,
-	root_url: string | null
+	server_url: string,
+	proxy_url: string | null
 ): string {
 	if (path == null) {
-		return root_url ? `/proxy=${root_url}file=` : `${root}/file=`;
+		return proxy_url ? `/proxy=${proxy_url}file=` : `${server_url}/file=`;
```

Comment:
I think this is wrong - path is never null since we set it in `upload` so we don't append the `/file=` part

Reply:
I still think it might be wrong, i thought that when i refactored but I didn't wanna change anything. I don't really understand when this would ever be useful, even if it is supported. Unless we use it to set some kind of base path for `sample_dir` in examples or something?

Reply:
I think this might be ok after all. I think images were not being loaded because we were using `url` instead of `path` as the img soruce